### PR TITLE
Created new page contributing-to-piwik-plugins

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Created new page contributing-to-piwik-plugins

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: docs/contributing-to-piwik-plugins.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+category: DevelopInDepth
+---
+
+# Contributing to Matomo Plugins
+
+### Commit messages to not trigger any release.
```
Comment (tsteur) [inline docs/contributing-to-piwik-plugins.md]: @AltamashShaikh might be a bit misleading headline as it suggests that a commit message might trigger a release?

File: docs/contributing-to-piwik-plugins.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+After your pull request is accepted and merged, the plugin needs to be released. Matomo tracks pushes to Matomo plugins using webhook.
```
Comment (tsteur) [inline docs/contributing-to-piwik-plugins.md]: Maybe you mean here we keep track of all merged commits into the main branch and notify the team once a week about all changes that were made in the previous week so everyone knows what's being released and to let us know we need to release a new version for that plugin?

File: docs/4.x/media-analytics/options.md [hunk 3]
```diff
@@ -192,7 +192,7 @@ _paq.push(['MediaAnalytics::scanForMedia']);
 Yes, the tracking of media events can be disabled by calling the following method:
 
 ```js
-_paq.push(['MediaAnalytics.disableTrackEvents']);
+_paq.push(['MediaAnalytics::disableTrackEvents']);
```
Comment (tsteur) [inline docs/4.x/media-analytics/options.md]: maybe this was added by accident? It should only be a single dot AFAIK as the method exists on the tracker (not on `Matomo.MediaAnalytics`).

**Review by tsteur — CHANGES_REQUESTED** [review]
Left a few comments @AltamashShaikh . We also need to make sure the page appears in a menu which you can add in https://github.com/matomo-org/developer-documentation/blob/live/app/helpers/Content/Category/DevelopInDepthCategory.php#L29

File: docs/4.x/media-analytics/options.md (hunk 3 above)
Comment (AltamashShaikh) [reply docs/4.x/media-analytics/options.md]: yes its a mistake reverted back.
Thanks

File: docs/contributing-to-piwik-plugins.md (hunk 1 above)
Comment (AltamashShaikh) [reply docs/contributing-to-piwik-plugins.md]: updated, please check if its still confusing

File: docs/contributing-to-piwik-plugins.md (hunk 2 above)
Comment (AltamashShaikh) [reply docs/contributing-to-piwik-plugins.md]: updated, please check if its still confusing

**Comment by tsteur** [comment]
This is now live https://developer.matomo.org/guides/contributing-to-piwik-plugins 🎉

**Comment by AltamashShaikh** [comment]
> This is now live https://developer.matomo.org/guides/contributing-to-piwik-plugins tada

:+1:  @tsteur

**Pull request merged**